src/jit/simdcodegenxarch.cpp

```diff
@@ -1158,7 +1158,7 @@ void CodeGen::genSIMDIntrinsic32BitConvert(GenTreeSIMD* simdNode)
             getEmitter()->emitIns_R_R_I(INS_pinsrw, emitTypeSize(TYP_INT), tmpReg, tmpIntReg, 3);
         }
 #endif
-        if (compiler->getSIMDInstructionSet() == InstructionSet_AVX)
+        if (compiler->getSIMDInstructionSet() == InstructionSet_AVX2)
         {
             inst_RV_RV(INS_vpbroadcastd, tmpReg, tmpReg, targetType, emitActualTypeSize(targetType));
         }
```

Comment:
@CarolEidt For example, once we distinguish `InstructionSet_AVX` and `InstructionSet_AVX2`, we will have to change here because `vpbroadcastd` is an AVX2 instruction.

Reply:
@CarolEidt ping?